### PR TITLE
Redirect user to Console for uninstallation of pubsub-stream-bigquery

### PR DESCRIPTION
**Pull request by tinaliang**
### Description

Regular uninstall flow:
Screen Shot 2020-08-28 at 12 28 45 PM

Uninstall flow for a pubsub-stream-bigquery extension
Screen Shot 2020-09-01 at 10 19 09 AM

### CODE REVIEW

File: src/commands/ext-uninstall.ts [hunk 1]
```diff
@@ -15,6 +18,26 @@ import { requirePermissions } from "../requirePermissions";
 import * as utils from "../utils";
 import * as logger from "../logger";
 
+marked.setOptions({
+  renderer: new TerminalRenderer(),
+});
+
+/**
+ * We do not currently support uninstalling extensions that require additional uninstall steps to be taken in the CLI. Direct them to the Console to uninstall the extension.
+ * - pubsub-stream-bigquery: requires calling a public HTTP uninstall function
```
Comment (laurenzlong) [inline src/commands/ext-uninstall.ts:27]: Please delete this comment line about "pubsub-stream-bigquery", don't need to go into implementation details here.

File: src/commands/ext-uninstall.ts [hunk 2]
```diff
@@ -15,6 +18,26 @@ import { requirePermissions } from "../requirePermissions";
...
+ *
+ * @param projectId id of the user's project
```
Comment (laurenzlong) [inline src/commands/ext-uninstall.ts:29]: capitalize ID in this line and below.

File: src/commands/ext-uninstall.ts [hunk 3]
```diff
@@ -15,6 +18,26 @@ import { requirePermissions } from "../requirePermissions";
...
+ * @param instanceId id of the extension instance
+ * @return a void Promise
+ */
+function consoleUninstallOnly(projectId: string, instanceId: string): Promise<void> {
+  const instanceURL = `https://console.firebase.google.com/project/${projectId}/extensions/instances/${instanceId}`;
+  const consoleUninstall =
```
Comment (samtstern) [inline src/commands/ext-uninstall.ts:35]: Is there any reason to use Markdown here?  I assume the renderer will only underline the link, or does it somehow make it more clickable?

Elsewhere in the CLI codebase I'd expect:
```ts
import * as clc from "cli-color";
// ...
const consoleUninstall = `This extension can only be uninstalled through the Firebase Console. Please visit ${clc.bold(underline(instanceUrl))} to uninstall this extension`
```

File: src/commands/ext-uninstall.ts [hunk 4]
```diff
@@ -15,6 +18,26 @@ import { requirePermissions } from "../requirePermissions";
 import * as utils from "../utils";
 import * as logger from "../logger";
 
+marked.setOptions({
```
Comment (samtstern) [inline src/commands/ext-uninstall.ts:21]: Does this modify a global state that will affect other uses of `marked` in the codebase?  If so is there a way to get an "instance"?

File: src/commands/ext-uninstall.ts (hunk 3 above)
Comment (tinaliang) [reply src/commands/ext-uninstall.ts:35]: On certain terminals, this turns into a clickable link

File: src/commands/ext-uninstall.ts (hunk 4 above)
Comment (tinaliang) [reply src/commands/ext-uninstall.ts:21]: This creates a reference instance: see https://marked.js.org/#/USING_ADVANCED.md

**Comment by annie-ryan** [comment]
> Option 1: Warning
> Screen Shot 2020-08-31 at 9 36 21 AM
> 

I agree that it should be presented more as info, rather than as if the user did something wrong.

I think having a space between "Required API ___ is enabled" and "This extension can only be installed through the Firebase console..." would help make it stand out.

Is there any other info we could add to explain to users why they must go to the console?

**Comment by tinaliang** [comment]
> > Option 1: Warning
> > Screen Shot 2020-08-31 at 9 36 21 AM
> 
> I agree that it should be presented more as info, rather than as if the user did something wrong.
> 
> I think having a space between "Required API ___ is enabled" and "This extension can only be installed through the Firebase console..." would help make it stand out.
> 
> Is there any other info we could add to explain to users why they must go to the console?

Updated the wording (see Description for screenshot), let me know what you think!

**Comment by tinaliang** [comment]
> > Updated the wording (see Description for screenshot), let me know what you think!
> 
> Maybe we can say something along the lines of "This extension has additional uninstall **checks** that are not currently supported by the CLI..."
> 
> I don't know if "checks" is the right word, but I don't want people going to the console and looking for extra steps to complete. Since the uninstall dialog is the same(ish).

I like that wording! Updated.

**Pull request merged**